### PR TITLE
pipeline tests - Ignore url.extension for ES < 8.14

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
Extension to https://github.com/elastic/integrations/pull/9623

Ignore the existence of an invalid `url.extension` field. Stack versions < 8.14 
had a bug that populated the field with bad data. After a package uses a minimum 
stack version of 8.14.0 then this addition to dynamic_fields can be removed.

This fixes errors like this which occur under v8.14.0+

```test case failed: Expected results are different from actual ones: --- want
+++ got
@@ -1797,7 +1797,6 @@
                 "preserve_duplicate_custom_fields"
             ],
             "url": {
-                "extension": "com/page",
                 "original": "www.example.com/page",
                 "path": "www.example.com/page"
             }
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/elasticsearch/pull/105689
- Relates https://github.com/elastic/integrations/issues/10649, https://github.com/elastic/integrations/issues/10628, https://github.com/elastic/integrations/issues/10621

### CODE REVIEW

File: packages/f5_bigip/data_stream/log/_dev/test/pipeline/test-common-config.yml [hunk 1]
```diff
@@ -2,3 +2,8 @@ fields:
   tags:
     - preserve_original_event
     - preserve_duplicate_custom_fields
+
+dynamic_fields:
+  # This can be removed after ES 8.14 is the minimum version.
+  # Relates: https://github.com/elastic/elasticsearch/pull/105689
+  url.extension: '^.*$'
```
Comment (efd6) [inline packages/f5_bigip/data_stream/log/_dev/test/pipeline/test-common-config.yml:9]: Final newlines.

**Review by efd6 — COMMENTED** [review]
Do we want version-next in the changelog?

**Comment by kcreddy** [comment]
> Do we want version-next in the changelog?

Following up on https://github.com/elastic/integrations/pull/9623, which also doesn't add changelog entries.

**Comment by efd6** [comment]
I'm fine with what is here.

**Review by efd6 — APPROVED** [review]

**Pull request merged**